java/ql/lib/semmle/code/java/security/WeakRandomnessQuery.qll

```diff
@@ -0,0 +1,162 @@
+/** Provides classes and predicates for reasoning about weak randomness. */
+
+import java
+import semmle.code.java.frameworks.Servlets
+import semmle.code.java.security.SensitiveActions
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.dataflow.ExternalFlow
+import semmle.code.java.security.RandomQuery
+
+/**
+ * The `java.util.Random` class.
+ */
+class TypeRandom extends RefType {
+  TypeRandom() { this.hasQualifiedName("java.util", "Random") }
+}
+
+/**
+ * A node representing a source of weak randomness.
+ *
+ * For example, use of `java.util.Random` or `java.lang.Math.random`.
+ */
+abstract class WeakRandomnessSource extends DataFlow::Node { }
```

Comment:
Note that we already have a [`RandomDataSource`](https://github.com/github/codeql/blob/7bea18773c5a5faa2ec1c9e9849dc7f27f784a60/java/ql/lib/semmle/code/java/security/RandomDataSource.qll#L10) definition. Maybe you could reuse it here, and improve it with your definitions if necessary.